File: tests/sample.cpp
// Sample C++ Code: Simple CLI To-Do App
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>

struct Task {
    std::string text;
    bool completed;

    Task(const std::string& t) : text(t), completed(false) {}
};

void displayTasks(const std::vector<Task>& tasks) {
    std::cout << "\nTo-Do List:" << std::endl;
    if (tasks.empty()) {
        std::cout << "  No tasks yet!" << std::endl;
    } else {
        for (size_t i = 0; i < tasks.size(); ++i) {
            std::cout << "  " << (i + 1) << ". [" << (tasks[i].completed ? "X" : " ") << "] " << tasks[i].text << std::endl;
        }
    }
}

void addTask(std::vector<Task>& tasks) {
    std::cout << "Enter a task: ";
    std::string taskText;
    std::getline(std::cin, taskText);
    if (!taskText.empty()) {
        tasks.emplace_back(taskText);
        std::cout << "Task added!" << std::endl;
    } else {
        std::cout << "Task cannot be empty." << std::endl;
    }
}

void deleteTask(std::vector<Task>& tasks) {
    std::cout << "Enter task number to delete: ";
    int taskNum;
    std::cin >> taskNum;
    std::cin.ignore(); // Consume newline
    if (taskNum > 0 && static_cast<size_t>(taskNum) <= tasks.size()) {
        tasks.erase(tasks.begin() + taskNum - 1);
        std::cout << "Task deleted!" << std::endl;
    } else {
        std::cout << "Invalid task number." << std::endl;
    }
}

void toggleTaskCompletion(std::vector<Task>& tasks) {
    std::cout << "Enter task number to toggle completion: ";
    int taskNum;
    std::cin >> taskNum;
    std::cin.ignore(); // Consume newline
    if (taskNum > 0 && static_cast<size_t>(taskNum) <= tasks.size()) {
        tasks[taskNum - 1].completed = !tasks[taskNum - 1].completed;
        std::cout << "Task status updated!" << std::endl;
    } else {
        std::cout << "Invalid task number." << std::endl;
    }
}

int main() {
    std::vector<Task> tasks;
    while (true) {
        std::cout << "\nOptions:" << std::endl;
        std::cout << "  1. View tasks" << std::endl;
        std::cout << "  2. Add task" << std::endl;
        std::cout << "  3. Delete task" << std::endl;
        std::cout << "  4. Toggle task completion" << std::endl;
        std::cout << "  5. Exit" << std::endl;

        std::cout << "Choose an option: ";
        int choice;
        std::cin >> choice;
        std::cin.ignore(); // Consume newline

        switch (choice) {
            case 1:
                displayTasks(tasks);
                break;
            case 2:
                addTask(tasks);
                break;
            case 3:
                deleteTask(tasks);
                break;
            case 4:
                toggleTaskCompletion(tasks);
                break;
            case 5:
                std::cout << "Goodbye!" << std::endl;
                return 0;
            default:
                std::cout << "Invalid choice. Please try again." << std::endl;
        }
    }
}

File: tests/sample.java
// Sample Java Code: Simple CLI To-Do App
import java.util.ArrayList;
import java.util.Scanner;

class Task {
    String text;
    boolean completed;

    Task(String text) {
        this.text = text;
        this.completed = false;
    }

    @Override
    public String toString() {
        return (completed ? "[X] " : "[ ] ") + text;
    }
}

public class ToDoApp {

    private static void displayTasks(ArrayList<Task> tasks) {
        System.out.println("\nTo-Do List:");
        if (tasks.isEmpty()) {
            System.out.println("  No tasks yet!");
        } else {
            for (int i = 0; i < tasks.size(); i++) {
                System.out.println("  " + (i + 1) + ". " + tasks.get(i));
            }
        }
    }

    private static void addTask(ArrayList<Task> tasks, Scanner scanner) {
        System.out.print("Enter a task: ");
        String taskText = scanner.nextLine().trim();
        if (!taskText.isEmpty()) {
            tasks.add(new Task(taskText));
            System.out.println("Task added!");
        } else {
            System.out.println("Task cannot be empty.");
        }
    }

    private static void deleteTask(ArrayList<Task> tasks, Scanner scanner) {
        System.out.print("Enter task number to delete: ");
        int taskNum = scanner.nextInt();
        scanner.nextLine(); // consume newline
        if (taskNum > 0 && taskNum <= tasks.size()) {
            tasks.remove(taskNum - 1);
            System.out.println("Task deleted!");
        } else {
            System.out.println("Invalid task number.");
        }
    }

    private static void toggleTaskCompletion(ArrayList<Task> tasks, Scanner scanner) {
        System.out.print("Enter task number to toggle completion: ");
        int taskNum = scanner.nextInt();
        scanner.nextLine(); // consume newline
        if (taskNum > 0 && taskNum <= tasks.size()) {
            Task task = tasks.get(taskNum - 1);
            task.completed = !task.completed;
            System.out.println("Task status updated!");
        } else {
            System.out.println("Invalid task number.");
        }
    }

    public static void main(String[] args) {
        ArrayList<Task> tasks = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\nOptions:");
            System.out.println("  1. View tasks");
            System.out.println("  2. Add task");
            System.out.println("  3. Delete task");
            System.out.println("  4. Toggle task completion");
            System.out.println("  5. Exit");

            System.out.print("Choose an option: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); // consume newline

            switch (choice) {
                case 1 -> displayTasks(tasks);
                case 2 -> addTask(tasks, scanner);
                case 3 -> deleteTask(tasks, scanner);
                case 4 -> toggleTaskCompletion(tasks, scanner);
                case 5 -> {
                    System.out.println("Goodbye!");
                    scanner.close();
                    return;
                }
                default -> System.out.println("Invalid choice. Please try again.");
            }
        }
    }
}
